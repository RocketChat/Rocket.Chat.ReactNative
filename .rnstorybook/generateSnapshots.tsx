import React from 'react';
import { composeStories } from '@storybook/react';
import { render } from '@testing-library/react-native';
import preview from './preview';
import { mockedStore } from '../app/reducers/mockedStore';
import { initStore } from '../app/lib/store/auxStore';

initStore(mockedStore);

export function generateSnapshots(stories: any) {
	describe('Story Snapshots:', () => {
		const composedStories = composeStories(stories, {
			decorators: preview.decorators
		});
		Object.entries(composedStories).forEach(([name, story]) => {
			test(`${name} should match snapshot`, () => {
				const rendered = render(React.createElement(story as React.ComponentType));
				expect(rendered.toJSON()).toMatchSnapshot();
			});
		});
	});
}
