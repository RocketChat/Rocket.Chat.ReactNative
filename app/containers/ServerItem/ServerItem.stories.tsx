import React from 'react';

import { themes } from '../../lib/constants';
import ServerItemComponent, { IServerItem } from '.';
import { ThemeContext, TSupportedThemes } from '../../theme';

export default {
	title: 'ServerItem'
};

const defaultItem = {
	name: 'Rocket.Chat',
	id: 'https://open.rocket.chat/',
	iconURL: 'https://open.rocket.chat/images/logo/android-chrome-512x512.png'
};

const ServerItem = ({
	item,
	theme = 'light',
	onPress = () => alert('Press'),
	onLongPress,
	hasCheck
}: {
	item?: Partial<IServerItem['item']>;
	theme?: TSupportedThemes;
	onPress?: IServerItem['onPress'];
	onLongPress?: IServerItem['onLongPress'];
	hasCheck?: IServerItem['hasCheck'];
}) => (
	<ThemeContext.Provider
		value={{
			theme,
			colors: themes[theme]
		}}
	>
		<ServerItemComponent item={{ ...defaultItem, ...item }} onPress={onPress} onLongPress={onLongPress} hasCheck={hasCheck} />
	</ThemeContext.Provider>
);

export const Content = () => (
	<>
		<ServerItem hasCheck />
		<ServerItem
			item={{
				name: 'Super Long Server Name in Rocket.Chat',
				id: 'https://superlongservername.tologintoasuperlongservername/'
			}}
		/>
		<ServerItem
			item={{
				id: 'https://stable.rocket.chat/'
			}}
		/>
	</>
);

export const Touchable = () => (
	<>
		<ServerItem onLongPress={() => alert('Long Press')} />
	</>
);

export const Themes = () => (
	<>
		<ServerItem theme={'light'} />
		<ServerItem theme={'dark'} />
		<ServerItem theme={'black'} />
	</>
);
