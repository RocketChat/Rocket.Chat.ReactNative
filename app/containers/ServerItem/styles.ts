import { StyleSheet } from 'react-native';

import sharedStyles from '../../views/Styles';

export const ROW_HEIGHT = 56;

export default StyleSheet.create({
	serverItemContainer: {
		flexDirection: 'row',
		alignItems: 'center',
		padding: 12
	},
	serverIcon: {
		width: 44,
		height: 44,
		borderRadius: 4,
		resizeMode: 'contain'
	},
	serverTextContainer: {
		flex: 1,
		flexDirection: 'column',
		justifyContent: 'center',
		paddingRight: 18,
		paddingLeft: 12
	},
	serverName: {
		fontSize: 18,
		...sharedStyles.textSemibold
	},
	serverUrl: {
		fontSize: 16,
		...sharedStyles.textRegular
	}
});
