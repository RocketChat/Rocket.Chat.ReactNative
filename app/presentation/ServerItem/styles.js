import { StyleSheet } from 'react-native';

import sharedStyles from '../../views/Styles';

export const ROW_HEIGHT = 56;

export default StyleSheet.create({
	serverItem: {
		height: ROW_HEIGHT,
		justifyContent: 'center'
	},
	serverItemContainer: {
		flexDirection: 'row',
		alignItems: 'center'
	},
	serverIcon: {
		width: 44,
		height: 44,
		marginHorizontal: 15,
		borderRadius: 4
	},
	serverTextContainer: {
		flex: 1,
		flexDirection: 'column',
		justifyContent: 'center'
	},
	serverName: {
		fontSize: 18,
		...sharedStyles.textSemibold
	},
	serverUrl: {
		fontSize: 15,
		...sharedStyles.textRegular
	}
});
