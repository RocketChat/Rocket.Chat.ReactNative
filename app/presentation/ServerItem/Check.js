import React from 'react';

import { CustomIcon } from '../../lib/Icons';
import styles from './styles';

const Check = React.memo(() => <CustomIcon style={styles.checkIcon} size={22} name='check' />);

export default Check;
