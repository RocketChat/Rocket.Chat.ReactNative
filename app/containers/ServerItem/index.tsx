import React from 'react';
// @ts-ignore // TODO: Remove on react-native update
import { Pressable, Text, View } from 'react-native';
import FastImage from 'react-native-fast-image';

import { IServerInfo } from '../../definitions';
import Check from '../Check';
import styles, { ROW_HEIGHT } from './styles';
import { themes } from '../../lib/constants';
import { isIOS } from '../../lib/methods/helpers';
import { useTheme } from '../../theme';

export { ROW_HEIGHT };

export interface IServerItem {
	item: IServerInfo;
	onPress(): void;
	onLongPress?(): void;
	hasCheck?: boolean;
}

const defaultLogo = require('../../static/images/logo.png');

const ServerItem = React.memo(({ item, onPress, onLongPress, hasCheck }: IServerItem) => {
	const { theme } = useTheme();
	return (
		<Pressable
			onPress={onPress}
			onLongPress={() => onLongPress?.()}
			testID={`rooms-list-header-server-${item.id}`}
			android_ripple={{ color: themes[theme].bannerBackground }}
			style={({ pressed }: { pressed: boolean }) => ({
				backgroundColor: isIOS && pressed ? themes[theme].bannerBackground : themes[theme].backgroundColor
			})}
		>
			<View style={styles.serverItemContainer}>
				{item.iconURL ? (
					<FastImage
						source={{
							uri: item.iconURL,
							priority: FastImage.priority.high
						}}
						// @ts-ignore TODO: Remove when updating FastImage
						defaultSource={defaultLogo}
						style={styles.serverIcon}
						onError={() => console.log('err_loading_server_icon')}
					/>
				) : (
					<FastImage source={defaultLogo} style={styles.serverIcon} />
				)}
				<View style={styles.serverTextContainer}>
					<Text numberOfLines={1} style={[styles.serverName, { color: themes[theme].titleText }]}>
						{item.name || item.id}
					</Text>
					<Text numberOfLines={1} style={[styles.serverUrl, { color: themes[theme].auxiliaryText }]}>
						{item.id}
					</Text>
				</View>
				{hasCheck ? <Check /> : null}
			</View>
		</Pressable>
	);
});

export default ServerItem;
