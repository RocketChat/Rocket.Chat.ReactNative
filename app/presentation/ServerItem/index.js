import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import FastImage from 'react-native-fast-image';

import Touch from '../../utils/touch';
import Check from '../../containers/Check';
import styles, { ROW_HEIGHT } from './styles';
import { themes } from '../../constants/colors';

export { ROW_HEIGHT };

const ServerItem = React.memo(({
	server, item, onPress, hasCheck, theme
}) => (
	<Touch
		onPress={onPress}
		style={[styles.serverItem, { backgroundColor: themes[theme].backgroundColor }]}
		testID={`rooms-list-header-server-${ item.id }`}
		theme={theme}
	>
		<View style={styles.serverItemContainer}>
			{item.iconURL
				? (
					<FastImage
						source={{
							uri: item.iconURL,
							priority: FastImage.priority.high
						}}
						defaultSource={{ uri: 'logo' }}
						style={styles.serverIcon}
						onError={() => console.log('err_loading_server_icon')}
					/>
				)
				: (
					<FastImage
						source={{ uri: 'logo' }}
						style={styles.serverIcon}
					/>
				)
			}
			<View style={styles.serverTextContainer}>
				<Text style={[styles.serverName, { color: themes[theme].titleText }]}>{item.name || item.id}</Text>
				<Text style={[styles.serverUrl, { color: themes[theme].auxiliaryText }]}>{item.id}</Text>
			</View>
			{item.id === server && hasCheck ? <Check theme={theme} /> : null}
		</View>
	</Touch>
));

ServerItem.propTypes = {
	onPress: PropTypes.func.isRequired,
	item: PropTypes.object.isRequired,
	hasCheck: PropTypes.bool,
	server: PropTypes.string,
	theme: PropTypes.string
};

export default ServerItem;
