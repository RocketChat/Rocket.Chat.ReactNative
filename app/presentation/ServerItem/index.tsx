import React from 'react';
// @ts-ignore
import { Pressable, Text, View } from 'react-native';
import FastImage from '@rocket.chat/react-native-fast-image';

import Check from '../../containers/Check';
import styles, { ROW_HEIGHT } from './styles';
import { themes } from '../../constants/colors';
import { isIOS } from '../../utils/deviceInfo';
import { withTheme } from '../../theme';

export { ROW_HEIGHT };

interface IServerItem {
	item: {
		id: string;
		iconURL: string;
		name: string;
	};
	onPress(): void;
	onLongPress?(): void;
	hasCheck?: boolean;
	theme?: string;
}

const defaultLogo = require('../../static/images/logo.png');

const ServerItem = React.memo(({ item, onPress, onLongPress, hasCheck, theme }: IServerItem) => (
	<Pressable
		onPress={onPress}
		onLongPress={() => onLongPress?.()}
		testID={`rooms-list-header-server-${item.id}`}
		android_ripple={{
			color: themes[theme!].bannerBackground
		}}
		style={({ pressed }: any) => ({
			backgroundColor: isIOS && pressed ? themes[theme!].bannerBackground : themes[theme!].backgroundColor
		})}>
		<View style={styles.serverItemContainer}>
			{item.iconURL ? (
				<FastImage
					source={{
						uri: item.iconURL,
						priority: FastImage.priority.high
					}}
					// @ts-ignore
					defaultSource={defaultLogo}
					style={styles.serverIcon}
					onError={() => console.log('err_loading_server_icon')}
				/>
			) : (
				<FastImage source={defaultLogo} style={styles.serverIcon} />
			)}
			<View style={styles.serverTextContainer}>
				<Text numberOfLines={1} style={[styles.serverName, { color: themes[theme!].titleText }]}>
					{item.name || item.id}
				</Text>
				<Text numberOfLines={1} style={[styles.serverUrl, { color: themes[theme!].auxiliaryText }]}>
					{item.id}
				</Text>
			</View>
			{hasCheck ? <Check theme={theme!} /> : null}
		</View>
	</Pressable>
));

export default withTheme(ServerItem);
