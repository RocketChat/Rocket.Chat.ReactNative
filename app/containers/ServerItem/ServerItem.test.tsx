import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './ServerItem.stories';

generateSnapshots(stories);
