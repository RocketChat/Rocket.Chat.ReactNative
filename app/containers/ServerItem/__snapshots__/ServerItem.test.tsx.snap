// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: Content should match snapshot 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFFFF",
      }
    }
    testID="server-item-https://open.rocket.chat/"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 12,
        }
      }
    >
      <ViewManagerAdapter_ExpoImage
        borderRadius={4}
        containerViewRef={"[React.ref]"}
        contentFit="contain"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        height={44}
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={
          [
            {
              "uri": "test-file-stub",
            },
          ]
        }
        source={
          [
            {
              "uri": "https://open.rocket.chat/images/logo/android-chrome-512x512.png",
            },
          ]
        }
        style={
          {
            "borderRadius": 4,
            "height": 44,
            "width": 44,
          }
        }
        transition={null}
        width={44}
      />
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingLeft": 12,
            "paddingRight": 18,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 18,
                "fontWeight": "600",
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Rocket.Chat
        </Text>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          https://open.rocket.chat/
        </Text>
      </View>
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#095AD2",
              "fontSize": NaN,
            },
            [
              {
                "lineHeight": NaN,
              },
              undefined,
            ],
            {
              "fontFamily": "custom",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î¤·
      </Text>
    </View>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFFFF",
      }
    }
    testID="server-item-https://superlongservername.tologintoasuperlongservername/"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 12,
        }
      }
    >
      <ViewManagerAdapter_ExpoImage
        borderRadius={4}
        containerViewRef={"[React.ref]"}
        contentFit="contain"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        height={44}
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={
          [
            {
              "uri": "test-file-stub",
            },
          ]
        }
        source={
          [
            {
              "uri": "https://open.rocket.chat/images/logo/android-chrome-512x512.png",
            },
          ]
        }
        style={
          {
            "borderRadius": 4,
            "height": 44,
            "width": 44,
          }
        }
        transition={null}
        width={44}
      />
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingLeft": 12,
            "paddingRight": 18,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 18,
                "fontWeight": "600",
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Super Long Server Name in Rocket.Chat
        </Text>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          https://superlongservername.tologintoasuperlongservername/
        </Text>
      </View>
    </View>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFFFF",
      }
    }
    testID="server-item-https://stable.rocket.chat/"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 12,
        }
      }
    >
      <ViewManagerAdapter_ExpoImage
        borderRadius={4}
        containerViewRef={"[React.ref]"}
        contentFit="contain"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        height={44}
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={
          [
            {
              "uri": "test-file-stub",
            },
          ]
        }
        source={
          [
            {
              "uri": "https://open.rocket.chat/images/logo/android-chrome-512x512.png",
            },
          ]
        }
        style={
          {
            "borderRadius": 4,
            "height": 44,
            "width": 44,
          }
        }
        transition={null}
        width={44}
      />
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingLeft": 12,
            "paddingRight": 18,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 18,
                "fontWeight": "600",
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Rocket.Chat
        </Text>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          https://stable.rocket.chat/
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`Story Snapshots: Themes should match snapshot 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#FFFFFF",
      }
    }
    testID="server-item-https://open.rocket.chat/"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 12,
        }
      }
    >
      <ViewManagerAdapter_ExpoImage
        borderRadius={4}
        containerViewRef={"[React.ref]"}
        contentFit="contain"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        height={44}
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={
          [
            {
              "uri": "test-file-stub",
            },
          ]
        }
        source={
          [
            {
              "uri": "https://open.rocket.chat/images/logo/android-chrome-512x512.png",
            },
          ]
        }
        style={
          {
            "borderRadius": 4,
            "height": 44,
            "width": 44,
          }
        }
        transition={null}
        width={44}
      />
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingLeft": 12,
            "paddingRight": 18,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 18,
                "fontWeight": "600",
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Rocket.Chat
        </Text>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "left",
              },
              {
                "color": "#6C727A",
              },
            ]
          }
        >
          https://open.rocket.chat/
        </Text>
      </View>
    </View>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#1F2329",
      }
    }
    testID="server-item-https://open.rocket.chat/"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 12,
        }
      }
    >
      <ViewManagerAdapter_ExpoImage
        borderRadius={4}
        containerViewRef={"[React.ref]"}
        contentFit="contain"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        height={44}
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={
          [
            {
              "uri": "test-file-stub",
            },
          ]
        }
        source={
          [
            {
              "uri": "https://open.rocket.chat/images/logo/android-chrome-512x512.png",
            },
          ]
        }
        style={
          {
            "borderRadius": 4,
            "height": 44,
            "width": 44,
          }
        }
        transition={null}
        width={44}
      />
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingLeft": 12,
            "paddingRight": 18,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 18,
                "fontWeight": "600",
                "textAlign": "left",
              },
              {
                "color": "#F2F3F5",
              },
            ]
          }
        >
          Rocket.Chat
        </Text>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "left",
              },
              {
                "color": "#9EA2A8",
              },
            ]
          }
        >
          https://open.rocket.chat/
        </Text>
      </View>
    </View>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#000000",
      }
    }
    testID="server-item-https://open.rocket.chat/"
  >
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 12,
        }
      }
    >
      <ViewManagerAdapter_ExpoImage
        borderRadius={4}
        containerViewRef={"[React.ref]"}
        contentFit="contain"
        contentPosition={
          {
            "left": "50%",
            "top": "50%",
          }
        }
        height={44}
        nativeViewRef={"[React.ref]"}
        onError={[Function]}
        onLoad={[Function]}
        onLoadStart={[Function]}
        onProgress={[Function]}
        placeholder={
          [
            {
              "uri": "test-file-stub",
            },
          ]
        }
        source={
          [
            {
              "uri": "https://open.rocket.chat/images/logo/android-chrome-512x512.png",
            },
          ]
        }
        style={
          {
            "borderRadius": 4,
            "height": 44,
            "width": 44,
          }
        }
        transition={null}
        width={44}
      />
      <View
        style={
          {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "center",
            "paddingLeft": 12,
            "paddingRight": 18,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 18,
                "fontWeight": "600",
                "textAlign": "left",
              },
              {
                "color": "#F2F3F5",
              },
            ]
          }
        >
          Rocket.Chat
        </Text>
        <Text
          numberOfLines={1}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "left",
              },
              {
                "color": "#9EA2A8",
              },
            ]
          }
        >
          https://open.rocket.chat/
        </Text>
      </View>
    </View>
  </View>,
]
`;

exports[`Story Snapshots: Touchable should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
    }
  }
  testID="server-item-https://open.rocket.chat/"
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "padding": 12,
      }
    }
  >
    <ViewManagerAdapter_ExpoImage
      borderRadius={4}
      containerViewRef={"[React.ref]"}
      contentFit="contain"
      contentPosition={
        {
          "left": "50%",
          "top": "50%",
        }
      }
      height={44}
      nativeViewRef={"[React.ref]"}
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      onProgress={[Function]}
      placeholder={
        [
          {
            "uri": "test-file-stub",
          },
        ]
      }
      source={
        [
          {
            "uri": "https://open.rocket.chat/images/logo/android-chrome-512x512.png",
          },
        ]
      }
      style={
        {
          "borderRadius": 4,
          "height": 44,
          "width": 44,
        }
      }
      transition={null}
      width={44}
    />
    <View
      style={
        {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
          "paddingLeft": 12,
          "paddingRight": 18,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 18,
              "fontWeight": "600",
              "textAlign": "left",
            },
            {
              "color": "#1F2329",
            },
          ]
        }
      >
        Rocket.Chat
      </Text>
      <Text
        numberOfLines={1}
        style={
          [
            {
              "backgroundColor": "transparent",
              "fontFamily": "Inter",
              "fontSize": 16,
              "fontWeight": "400",
              "textAlign": "left",
            },
            {
              "color": "#6C727A",
            },
          ]
        }
      >
        https://open.rocket.chat/
      </Text>
    </View>
  </View>
</View>
`;
